### PR TITLE
initial styling for ForceNewPassword

### PR DESCRIPTION
**Pull request by eddiekeller**
*Issue #, if available:*

https://app.asana.com/0/1200881947262804/1200952565427106/f

*Description of changes:*

Initial styling for the `ForceNewPassword` screen. This adds a new `confirm_password` field to the screen. As a result, I also modified the state machine to handle validating that the passwords match each other, identical to the `SignUp` screen (literally copy/pasted most of it).

This PR *might* break functionality in Vue and Angular on the `ForceNewPassword` screen due to changes in the state machine.

Screen Shot 2021-09-10 at 11 32 11 AM
Screen Shot 2021-09-10 at 11 33 03 AM
Screen Shot 2021-09-10 at 11 33 40 AM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/e2e/cypress/integration/ui/components/authenticator/sign-in-force-new-password/sign-in-force-new-password.steps.ts [hunk 1]
```diff
@@ -10,7 +10,11 @@ And('I type in the phone number {string}', (phone: string) => {
 });
 
 And('I type in the password {string}', (password: string) => {
-  cy.findByPlaceholderText(/password/i).type(Cypress.env(password));
+  cy.findByPlaceholderText(/^password$/i).type(Cypress.env(password));
+});
+
+And('I confirm the password {string}', (password: string) => {
+  cy.findByPlaceholderText(/^confirm_password$/i).type(Cypress.env(password));
```
Comment (slaymance) [inline packages/e2e/cypress/integration/ui/components/authenticator/sign-in-force-new-password/sign-in-force-new-password.steps.ts:17]: These tests are skipped but I think the regex for the placeholder selector should be `/^confirm password$/i`.

File: packages/ui/src/machines/actors/auth/signIn.ts [hunk 2]
```diff
@@ -401,6 +451,11 @@ export const signInActor = createMachine<SignInContext, AuthEvent>(
 
         return result;
       },
+      async validateFields(context, _event) {
+        const { formValues } = context;
+        const validators = [passwordMatches]; // this can contain custom validators too
+        return runValidators(formValues, validators);
+      },
```
Comment (wlee221) [inline packages/ui/src/machines/actors/auth/signIn.ts:458]: Can we share this in `../../actions`?

**Review by wlee221 — APPROVED** [review]
All lgtm! We should consider abstracting "form with validation" interface so that we can reuse it across subcomponents. Could be its own actor, or just a dynamic object. But I'm digressing again, code changes here lgtm 🚢

File: packages/ui/src/machines/actors/auth/signIn.ts (hunk 2 above)
Comment (eddiekeller) [reply packages/ui/src/machines/actors/auth/signIn.ts:458]: I wonder how that works with services? I'll take a look at this for a future PR!

**Review by slaymance — APPROVED** [review]
Left a small comment for the e2e test selector.

File: packages/ui/src/machines/actors/auth/signIn.ts (hunk 2 above)
Comment (wlee221) [reply packages/ui/src/machines/actors/auth/signIn.ts:458]: Oh shoot, yeah this is services 😓  my bad. We should be able to share those anyways, but I agree and this is not imminent.

File: packages/e2e/cypress/integration/ui/components/authenticator/sign-in-force-new-password/sign-in-force-new-password.steps.ts (hunk 1 above)
Comment (eddiekeller) [reply packages/e2e/cypress/integration/ui/components/authenticator/sign-in-force-new-password/sign-in-force-new-password.steps.ts:17]: Good point! It should be. I'll have another PR following this one pretty quickly so I'll make the change there.

**Pull request merged**